omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -31,83 +33,123 @@ function populateReportsDropdown() {
         }
 
         jQuery('#reportsSelectContainer').html("");
-
         jQuery('#reportsSelectContainer').append(reportSelect);
 
         jQuery("#reportSelect").hide().fadeIn("slow");
 
         reports = data.results;
     });
 }
-
+function hideDiv(){
+    jQuery('#dailyPicker').hide();
+    jQuery('#weeklyPicker').hide();
+    jQuery('#monthlyPicker').hide();
+    jQuery('#yearlyPicker').hide();
+}
 function onMappingSelect() {
+    selectedMapping = mappings[jQuery('#mappingSelect').val()];
     // clear the date
     document.getElementById('custom-range-option').checked = false;
     jQuery('#un-recognized-period').html('');
     toggleCustomRangeCheckbox(false);
     jQuery('#periodSelector').val("");
     jQuery('#periodSelector').monthpicker('destroy');
     jQuery('#periodSelector').datepicker('destroy');
-
     jQuery('#ui-datepicker-div > table > tbody > tr').die('mousemove');
     jQuery('.ui-datepicker-calendar tr').die('mouseleave');
-
     weekStartDate = null;
     weekEndDate = null;
+    hideDiv();
+    switch(selectedMapping.periodType){
+        case 'Daily':
+            intializeDailyPicker();
+            break;
+        case 'Weekly':
+            initializeWeeklyPicker();
+            break;
+        case 'Monthly':
+            initaialzieMonthlyPicker();
+            break
+        case 'Yearly':
+            initializeYearlyPicker();
+            break;
+    }
+}
 
-    // get periodType
-    var selectedPeriodType = mappings.filter(function (v) {
-        return v.created == jQuery('#mappingSelect').val();
-    })[0].periodType;
-
-    if (selectedPeriodType == undefined)
-        return;
+function initializeYearlyPicker(){
+    const currentYear = moment().year();
+    jQuery('#yearlyPicker').show();
+    jQuery('#yearSelector').attr("max", currentYear-1);
+    jQuery('#yearSelector').val(currentYear-1);
+    onYearChange();
+    document.getElementById("datepicker").addEventListener("change", function() {
+        selectedDateRange = this.value;
+    });
```

Comment:
Seems we don't need this listener because the yearly input has already linked with `onYearChange()` function